### DISCUSSION TITLE
Use python-pptx for customizable reports in the form of Powerpoint Presentations

### DISCUSSION THREAD

**Discussion by GRPlan** [pyroll-project/pyroll-core#223] [Ideas]
Attached a quick demonstration for the idea to use python-pptx for customizable reports in the form of powerpoint-Presentations.

[ReportPPTX.zip](https://github.com/user-attachments/files/15542820/ReportPPTX.zip)

It uses
https://pypi.org/project/python-pptx/

I see the advantage that the template can be easily adjusted to corporate (user-specific) powerpoint-Templates.

**Comment by sag-kalibreur** [comment]
I have seen that you are now using the stand orientation from the tools.xlsx, which is absolutely my favourite :-)
The PowerPoint template is another good idea (the plots still need to be adjusted slightly).
The question is which standard should be established? An HTML report, PDF print or PowerPoint. PowerPoint would have the great advantage of being customisable to a corporate design and PDF printing is also possible.

However, I get an attribute error when executing **write_image** to write to the pptx. I have not yet been able to find a quick solution to this problem:

AttributeError: **'Figure'** object has no attribute **'write_image'**

![image](https://github.com/pyroll-project/pyroll-core/assets/144391129/d8124b79-0768-4231-a8e6-0bef1b496431)

Do you have any idea what could be the reason, perhaps an incompatible version?

**Reply by GRPlan** [reply]
Check pyroll.core version.
My test was on '2.1.5'

**Reply by sag-kalibreur** [reply]
Test was run with core 2.1.6

**Reply by GRPlan** [reply]
Reason is used Plotting-Backend

pr.PLOTTING_BACKEND is 'matplotlib' then use
`fig.savefig(image_stream)`
instead

`fig_write_image(image_stream)` works for pr.PLOTTING_BACKEND = 'plotly'

**Reply by sag-kalibreur** [reply]
Yes my used plotting backend is matplotlib, `fig.savefig(image_stream)` works, thanks
